core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala

```diff
@@ -206,6 +206,39 @@ class KafkaConfigTest {
     assertBadConfigContainingMessage(props, "Each listener must have a different name")
   }
 
+  @Test
+  def testIPv4AndIPv6SamePortListeners(): Unit = {
```

Comment:
Can we add a test with a hostname and an IPv6 on the same port?
Soemthing like:
```
props.put(KafkaConfig.ListenersProp, "PLAINTEXT://apache.org:9092,PLAINTEXT://[::1]:9092")
caught = assertThrows(classOf[IllegalArgumentException], () => KafkaConfig.fromProps(props))
assertTrue(caught.getMessage.contains("Each listener must have a different port"))

props.put(KafkaConfig.ListenersProp, "PLAINTEXT://apache.org:9092,PLAINTEXT://127.0.0.1:9092")
caught = assertThrows(classOf[IllegalArgumentException], () => KafkaConfig.fromProps(props))
assertTrue(caught.getMessage.contains("Each listener must have a different port"))
```

Reply:
Test added and I can confirm it passes locally